// viio-project-tools/dist/crud/BalanceManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceManager = void 0;
const CrudServiceManager_1 = require("./CrudServiceManager");
class BalanceManager extends CrudServiceManager_1.CrudServiceManager {
    constructor(url) {
        super();
        this.url = url;
    }
    findById(id) {
        return this.sendFinding(`${this.url}/balances?id=${id.id}`, id.config);
    }
    find(details) {
        return this.sendFinding(`${this.url}/balances?userId=${details.userId}&assetId=${details.assetId}&sourceId=${details.sourceId}`, details.config);
    }
    create(details) {
        return this.sendCreation(`${this.url}/balances`, { userId: details.userId, assetId: details.assetId, sourceId: details.sourceId, balance: 0, temporalBalance: 0 }, details.config);
    }
    modify(balance) {
        return this.sendModification(`${this.url}/balances/${balance.balanceId}`, balance, balance.config);
    }
    destroy(balance) {
        return this.sendDestuction(`${this.url}/balances/${balance.balanceId}`, balance.config);
    }
}
exports.BalanceManager = BalanceManager;
