// viio-project-tools/dist/crud/DigitalCurrencyManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DigitalCurrencyManager = void 0;
const CrudServiceManager_1 = require("./CrudServiceManager");
;
class DigitalCurrencyManager extends CrudServiceManager_1.CrudServiceManager {
    constructor(url) {
        super();
        this.url = url;
    }
    findById(id) {
        return this.sendFinding(`${this.url}/assets?id=${id.id}`, id.config);
    }
    find(details) {
        return this.sendFinding(`${this.url}/assets?ticker=${details.assetId}`, details.config);
    }
    create(details) {
        return this.sendCreation(`${this.url}/assets`, {
            name: { en: details.symbol },
            ticker: details.assetId,
            type: "DigitalCurrency",
            details: {
                decimals: details.decimals,
                chain: details.chain,
                symbol: details.symbol
            }
        }, details.config);
    }
    modify(asset) {
        return this.sendModification(`${this.url}/assets/${asset.assetId}`, asset, asset.config);
    }
    destroy(asset) {
        return this.sendDestuction(`${this.url}/assets/${asset.assetId}`, asset.config);
    }
}
exports.DigitalCurrencyManager = DigitalCurrencyManager;
