// viio-project-tools/dist/crud/TransactionTypeManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionTypeManager = void 0;
const CrudServiceManager_1 = require("./CrudServiceManager");
class TransactionTypeManager extends CrudServiceManager_1.CrudServiceManager {
    constructor(url) {
        super();
        this.url = url;
    }
    find(details) {
        return this.sendFinding(`${this.url}/transactionTypes?type=${details.type}`, details.config);
    }
    findById(id) {
        return this.sendFinding(`${this.url}/transactionTypes?id=${id.id}`, id.config);
    }
    create(details) {
        var _a;
        return this.sendCreation(`${this.url}/transactionTypes`, {
            type: details.type,
            description: (_a = details.description) !== null && _a !== void 0 ? _a : { en: details.type }
        }, details.config);
    }
    modify(type) {
        return this.sendModification(`${this.url}/transactionTypes/${type.transactionTypeId}`, {
            type: type.type,
            description: type.description
        }, type.config);
    }
    destroy(type) {
        return this.sendDestuction(`${this.url}/transactionTypes/${type.transactionTypeId}`, type.config);
    }
}
exports.TransactionTypeManager = TransactionTypeManager;
