// viio-project-tools/dist/crud/TransactionStatusManager.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionStatusManager = void 0;
const CrudServiceManager_1 = require("./CrudServiceManager");
class TransactionStatusManager extends CrudServiceManager_1.CrudServiceManager {
    constructor(url) {
        super();
        this.url = url;
    }
    find(details) {
        return this.sendFinding(`${this.url}/transactionStatus?status=${details.status}`, details.config);
    }
    findById(id) {
        return this.sendFinding(`${this.url}/transactionStatus?id=${id.id}`, id.config);
    }
    create(details) {
        var _a;
        return this.sendCreation(`${this.url}/transactionStatus`, {
            status: details.status,
            description: (_a = details.description) !== null && _a !== void 0 ? _a : { en: details.status }
        }, details.config);
    }
    modify(status) {
        return this.sendModification(`${this.url}/transactionStatus/${status.transactionStatusId}`, {
            status: status.status,
            description: status.description
        }, status.config);
    }
    destroy(status) {
        return this.sendDestuction(`${this.url}/transactionStatus/${status.transactionStatusId}`, status.config);
    }
}
exports.TransactionStatusManager = TransactionStatusManager;
