// viio-project-tools/dist/definitions/TransactionDefinitions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferType = exports.TransactionType = exports.TransactionStatus = void 0;
var TransactionStatus;
(function (TransactionStatus) {
    TransactionStatus["APPROVED"] = "APPROVED";
    TransactionStatus["SUBMITTED"] = "SUBMITTED";
    TransactionStatus["INSUFFICIENT_FUNDS"] = "INSUFFICIENT_FUNDS";
    TransactionStatus["CONDUIT_INSUFFICIENT_FUNDS"] = "CONDUIT_INSUFFICIENT_FUNDS";
    TransactionStatus["COMPLETED"] = "COMPLETED";
    TransactionStatus["BLOCKED"] = "BLOCKED";
    TransactionStatus["FAILED"] = "FAILED";
    TransactionStatus["IN_TREASURY"] = "IN_TREASURY";
    TransactionStatus["IN_CONDUIT_TREASURY"] = "IN_CONDUIT_TREASURY";
    TransactionStatus["RECEIVED"] = "RECEIVED";
    TransactionStatus["REJECTED"] = "REJECTED";
    TransactionStatus["REFUND"] = "REFUND";
})(TransactionStatus || (exports.TransactionStatus = TransactionStatus = {}));
var TransactionType;
(function (TransactionType) {
    TransactionType["DEPOSIT"] = "DEPOSIT";
    TransactionType["EXTERNAL_DEPOSIT"] = "EXTERNAL_DEPOSIT";
    TransactionType["EXTERNAL_WITHDRAWAL"] = "EXTERNAL_WITHDRAWAL";
    TransactionType["REFUND"] = "REFUND";
    TransactionType["TRANSFER"] = "TRANSFER";
    TransactionType["WITHDRAWAL"] = "WITHDRAWAL";
    TransactionType["OTC_DEPOSIT"] = "OTC_DEPOSIT";
    TransactionType["OTC_WITHDRAWAL"] = "OTC_WITHDRAWAL";
    TransactionType["CONDUIT_DEPOSIT"] = "CONDUIT_DEPOSIT";
    TransactionType["CONDUIT_WITHDRAWAL"] = "CONDUIT_WITHDRAWAL";
})(TransactionType || (exports.TransactionType = TransactionType = {}));
var TransferType;
(function (TransferType) {
    TransferType["SENDER"] = "sender";
    TransferType["RECEIVER"] = "receiver";
})(TransferType || (exports.TransferType = TransferType = {}));
